### PR TITLE
Move integration tests to cli/tests/

### PR DESCRIPTION
**Pull request by ry**
This ensures the deno executable is properly created before running the integration tests.

Also allows deno_cli to be used as a lib. Docs are now properly generated: https://docs.rs/deno_cli/0.18.4/deno_cli/ 

Towards https://github.com/denoland/deno/issues/2933
Prep for https://github.com/denoland/deno/pull/2955

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 js/flatbuffers.js
 tests/error_syntax.js
```
Comment (piscisaureus) [inline .prettierignore]: Remove

File: .prettierignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
 tests/badly_formatted.js
```
Comment (piscisaureus) [inline .prettierignore]: Remove

File: cli/tests/integration_tests.rs [hunk 3]
```diff
@@ -1,5 +1,8 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-use crate::colors::strip_ansi_codes;
+#[macro_use]
+extern crate log;
+
+use deno_cli::colors::strip_ansi_codes;
```
Comment (piscisaureus) [inline cli/tests/integration_tests.rs]: Just curious, why the change from `crate::colors` to `deno_cli::colors`?
It's still the same crate right?

File: js/stat_test.ts [hunk 4]
```diff
@@ -54,7 +54,7 @@ testPerm({ read: true }, async function lstatSyncSuccess(): Promise<void> {
   assert(!modulesInfo.isDirectory());
   assert(modulesInfo.isSymlink());
 
-  const testsInfo = Deno.lstatSync("tests");
+  const testsInfo = Deno.lstatSync("website");
```
Comment (piscisaureus) [inline js/stat_test.ts]: Maybe update the variable name too?
As far as I'm concerned it could be just `const info2 = ...`.
But **tests**Info = Deno.lstatSync("**website**") is kinda weird.

File: cli/tests/integration_tests.rs (hunk 3 above)
Comment (ry) [reply cli/tests/integration_tests.rs]: No, this is another crate that gets built after deno_cli (crucially that's when the deno binary is available)

File: js/stat_test.ts (hunk 4 above)
Comment (ry) [reply js/stat_test.ts]: done

File: .prettierignore (hunk 1 above)
Comment (ry) [reply .prettierignore]: done

File: .prettierignore (hunk 2 above)
Comment (ry) [reply .prettierignore]: done

File: js/stat_test.ts [hunk 5]
```diff
@@ -138,7 +138,7 @@ testPerm({ read: true }, async function lstatSuccess(): Promise<void> {
   assert(!modulesInfo.isDirectory());
   assert(modulesInfo.isSymlink());
 
-  const testsInfo = await Deno.lstat("tests");
+  const testsInfo = await Deno.lstat("website");
```
Comment (piscisaureus) [inline js/stat_test.ts]: ditto

Comment (ry) [reply js/stat_test.ts]: done
